### DISCUSSION TITLE
gns3 pid and gns3 profiles

### DISCUSSION THREAD

**Discussion by josephmhiggins** [GNS3/gns3-server#2122] [General]
1. re gns3 pid: I would have to find what Jerry Grossman wrote about pid. Using persistence will allow forensics to be done on syslog entries, but...I have the pid when I start gns3 in my script (right now I use Start-Process with the -PassThru option in powershell, but I think, but I am not sure, I will switch to the System.Diagnostics.Process
I am not sure how that will shake out yet.
2. I really do not like the gns3 developers implementation of Profiles...........but only new gns3 users seem to have problems.  And new gns3 users do not profiles yet AND even though I have all kinds of different named GNS3 VMs (i.e. gns3 vm 2.2.31, gns3 vm 2.2.34), my scripts look like they only like a gns3 vm that is named GNS3 VM, AND I do not like get any data from a gns3 profile config file; I only like asking the gns3 controller for information of a currently running gns3.

So, this thread,  https://github.com/GNS3/gns3-server/discussions/2118, https://github.com/GNS3/gns3-server/discussions/2120, https://github.com/GNS3/gns3-server/issues/2117, covers everything that I can think about the Gns3 configuration directory.

**Comment by josephmhiggins** [comment]
Oh yeah, gns3 runs on a Mac somehow. I do not know anything about a Mac, so.....on Windows a thing like pid just could be extraneous information, but on a Mac it could be something that is a "must have".

**Comment by eantowne** [comment]
Max OS X is a Unix compliant OS, so yes the PID exists and is important.

On Sat, Oct 1, 2022 at 08:52 josephhiggins ***@***.***> wrote:

> Oh yeah, gns3 runs on a Mac somehow. I do not know anything about a Mac,
> so.....on Windows a thing like pid just could be extraneous information,
> but on a Mac it could be something that is a "must have".
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GNS3/gns3-server/discussions/2122#discussioncomment-3777964>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APZX5IZ4AHW7YYMKOU27I2TWBAXZZANCNFSM6AAAAAAQ2NR56Q>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Ean L. Towne

***@***.***

**Comment by josephmhiggins** [comment]
This falls under Ean's wait for gns3 version 3.0
This can be closed.
Basically, Ean does "Greenfield" deployments and I want to do that, hopefully, by next year.
